File: fixtures/runtime-parity/lru-nested-field-write-depth2/solution.js
class Node {
  constructor(value) {
    this.value = value;
    this.prev = null;
    this.next = null;
  }
}

class Solution {
  constructor() {
    this.head = new Node(-1);
    this.tail = new Node(-1);
    this.head.next = this.tail;
    this.tail.prev = this.head;
  }

  solve(value) {
    const node = new Node(value);
    this.head.next.prev = node;
    this.head.next = node;
    node.prev = this.head;
    node.next = this.tail;
    return this.head.next.value;
  }
}

File: fixtures/runtime-parity/lru-nested-field-write-depth2/solution.py
class Node:
    def __init__(self, value):
        self.value = value
        self.prev = None
        self.next = None

class Solution:
    def __init__(self):
        self.head = Node(-1)
        self.tail = Node(-1)
        self.head.next = self.tail
        self.tail.prev = self.head

    def solve(self, value):
        node = Node(value)
        self.head.next.prev = node
        self.head.next = node
        node.prev = self.head
        node.next = self.tail
        return self.head.next.value
